toctocbin/AndroidArchitecture/app/src/main/java/com/test/androidarchitecture/ui/ticker/TickerPresenter.kt

```diff
@@ -0,0 +1,66 @@
+package com.test.androidarchitecture.ui.ticker
+
+import com.test.androidarchitecture.R
+import com.test.androidarchitecture.data.Ticker
+import com.test.androidarchitecture.data.TickerFormat
+import com.test.androidarchitecture.data.source.UpbitRepository
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import java.text.DecimalFormat
+import java.text.NumberFormat
+import java.util.*
+
+class TickerPresenter(private val view: TickerContract.View) : TickerContract.Presenter {
+
+    private val upbitRepository by lazy { UpbitRepository }
+    private val disposables by lazy { CompositeDisposable() }
+
+
+    override fun getTicker(marketSearch: String) {
+        disposables.add(upbitRepository.getTicker(marketSearch)
+            .map { list ->
+                list.asSequence()
+                    .map { getCoinFormat(it) }
+                    .toList()
+            }
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(
+                {
+                    view.setTickerData(it)
+                }, {
+                    view.showToast(it.message.toString())
+                }
+            ))
+
+    }
+
+    override fun disposablesClear() {
+        disposables.clear()
+    }
+
+    private fun getCoinFormat(ticker: Ticker): TickerFormat {
+        val marketName: String = ticker.market.substringAfter("-")
```

Comment:
초기화 값의 타입이 명확한 경우, changeRate 변수와 같이 타입을 생략하셔도 됩니다.
```suggestion
        val marketName = ticker.market.substringAfter("-")
```

Reply:
1e3e95f

제거하였습니다.